osquery/tables/system/linux/deb_packages.cpp

```diff
@@ -0,0 +1,185 @@
+    {"Priority", f_priority, w_priority},
+    {"Section", f_section, w_section},
+    {"Installed-Size", f_charfield, w_charfield, PKGIFPOFF(installedsize)},
+    {"Origin", f_charfield, w_charfield, PKGIFPOFF(origin)},
+    {"Maintainer", f_charfield, w_charfield, PKGIFPOFF(maintainer)},
+    {"Bugs", f_charfield, w_charfield, PKGIFPOFF(bugs)},
+    {"Architecture", f_architecture, w_architecture},
+    {"Multi-Arch", f_multiarch, w_multiarch, PKGIFPOFF(multiarch)},
+    {"Source", f_charfield, w_charfield, PKGIFPOFF(source)},
+    {"Version", f_version, w_version, PKGIFPOFF(version)},
+    {"Revision", f_revision, w_null},
+    {"Config-Version", f_configversion, w_configversion},
+    {"Replaces", f_dependency, w_dependency, dep_replaces},
+    {"Provides", f_dependency, w_dependency, dep_provides},
+    {"Depends", f_dependency, w_dependency, dep_depends},
+    {"Pre-Depends", f_dependency, w_dependency, dep_predepends},
+    {"Recommends", f_dependency, w_dependency, dep_recommends},
+    {"Suggests", f_dependency, w_dependency, dep_suggests},
+    {"Breaks", f_dependency, w_dependency, dep_breaks},
+    {"Conflicts", f_dependency, w_dependency, dep_conflicts},
+    {"Enhances", f_dependency, w_dependency, dep_enhances},
+    {"Conffiles", f_conffiles, w_conffiles},
+    {"Filename", f_filecharf, w_filecharf, FILEFOFF(name)},
+    {"Size", f_filecharf, w_filecharf, FILEFOFF(size)},
+    {"MD5sum", f_filecharf, w_filecharf, FILEFOFF(md5sum)},
+    {"MSDOS-Filename", f_filecharf, w_filecharf, FILEFOFF(msdosname)},
+    {"Description", f_charfield, w_charfield, PKGIFPOFF(description)},
+    {"Triggers-Pending", f_trigpend, w_trigpend},
+    {"Triggers-Awaited", f_trigaw, w_trigaw},
+    /* Note that aliases are added to the nicknames table. */
+    {NULL}};
+
+Row extractDebPackageInfo(const struct pkginfo *pkg,
+                          const struct pkgbin *pkgbin) {
+  Row r;
+  const struct fieldinfo *fip;
+
+  struct varbuf vb;
+  varbuf_init(&vb, 20);
+  for (fip = fieldinfos; fip->name; fip++) {
+    fip->wcall(&vb, pkg, pkgbin, fw_printheader, fip);
+
+    std::string line = vb.string();
+    if (!line.empty()) {
+      std::size_t separator_position = line.find(':');
+      std::string key = line.substr(0, separator_position);
+      std::string value = line.substr(separator_position + 1, line.length());
+      auto it = kFieldMappings.find(key);
+      if (it != kFieldMappings.end()) {
+        boost::algorithm::trim(value);
+        r[it->second] = value;
+        }
+    }
+    varbuf_reset(&vb);
+  }
+  varbuf_destroy(&vb);
+
+  return r;
+}
+
+QueryData genDebs(QueryContext &context) {
+  QueryData results;
+  struct pkg_array array;
+  struct pkginfo *pkg;
+  int i;
+
+  dpkg_setup("dpkg", msdbrw_readonly, &array);
+
+  for (i = 0; i < array.n_pkgs; i++) {
+    pkg = array.pkgs[i];
+
+    if (pkg->status == pkg->stat_notinstalled) {
+      continue;
+    }
+
+    results.push_back(extractDebPackageInfo(pkg, &pkg->installed));
```

Comment:
Why pass `pkg` and `&pkg->installed` here?

Also, you can pass `&results` as a param and avoid copying the returned Row, like: https://github.com/facebook/osquery/blob/master/osquery/tables/system/crontab.cpp#L58 just a nit.